### ISSUE TITLE
"Upsert" Insights account ID into our models

### ISSUE THREAD

**Issue by katherine-black** [cloudigrade/cloudigrade#595]
In GitLab by @infinitewarp on Jun 7, 2019, 13:44

## Summary

As a user on the Insights platform, I want cloudigrade to remember my Insights account ID so it can use it to communicate on my behalf with other back-end Insights services (such as Sources API).

## Acceptance Criteria

- [ ] Verify that when a request authorized by 3scale arrives, cloudigrade stores the Insights account ID somewhere related to the User model.
- [ ] Verify that if the User already exists (e.g. it was created before we started tracking account ID), the model is gracefully updated with the new account ID value.
- [ ] Verify that if this change requires a new User-derived model, all existing User objects are automatically migrated.

## Assumptions and Questions

- We need this so async/periodic tasks can work with Sources API (and presumably others in the future) when we do not have an HTTP identity header.
  - Working with Sources API requires an `'X-RH-IDENTITY'` HTTP header with at least the following amount of content base64-encoded:
    ```
    {"identity":{"account_number": "5910538"}}
    ```
- Given this example pretty-printed and decoded header from 3scale, what we need to keep (at a minimum) is identity.account_number.
  ```
  {
      "entitlements": {
          "hybrid_cloud": {
              "is_entitled": true
          },
          "insights": {
              "is_entitled": true
          },
          "openshift": {
              "is_entitled": true
          },
          "smart_management": {
              "is_entitled": true
          }
      },
      "identity": {
          "account_number": "540155",
          "internal": {
              "auth_time": 1899,
              "auth_type": "basic-auth",
              "org_id": "1979710"
          },
          "type": "User",
          "user": {
              "email": "ccabrera+qa@redhat.com",
              "first_name": "Charles",
              "is_active": true,
              "is_internal": false,
              "is_org_admin": false,
              "last_name": "Cabrera",
              "locale": "en_US",
              "username": "cdcabrera"
          }
      }
  }
  ```
- While we adding support for this one field, do we think there's value in keeping a copy of more things from the incoming 3scale auth header?
- It looks like in ci and prod, multiple users are sharing the same "account ID" (TBD if that's the right term). So, whatever we implement should _not_ enforce uniqueness of this value across different Users.

**Comment by katherine-black** [comment]
In GitLab by @infinitewarp on Jun 11, 2019, 11:22

Closing this in favor of bigger relationship changes defined in https://gitlab.com/cloudigrade/cloudigrade/issues/593

**Comment by katherine-black** [comment]
In GitLab by @infinitewarp on Jun 11, 2019, 11:22

closed
